x-pack/plugins/maps/public/components/layer_addpanel/index.js

```diff
@@ -7,43 +7,46 @@
 import { connect } from 'react-redux';
 import { AddLayerPanel } from './view';
 import { getFlyoutDisplay, updateFlyout, FLYOUT_STATE } from '../../store/ui';
-import { getTemporaryLayers, getMapColors } from '../../selectors/map_selectors';
+import { getSelectedLayer, getMapColors } from '../../selectors/map_selectors';
 import { getInspectorAdapters } from '../../store/non_serializable_instances';
 import {
-  addLayer,
   removeLayer,
-  clearTemporaryLayers,
+  setTransientLayer,
+  addLayer,
   setSelectedLayer,
+  removeTransientLayer
 } from '../../actions/store_actions';
-import _ from 'lodash';
 
 function mapStateToProps(state = {}) {
-
-  function isLoading() {
-    const tmp = getTemporaryLayers(state);
-    return tmp.some((layer) => layer.isLayerLoading());
-  }
+  const selectedLayer = getSelectedLayer(state);
   return {
     inspectorAdapters: getInspectorAdapters(state),
     flyoutVisible: getFlyoutDisplay(state) !== FLYOUT_STATE.NONE,
-    layerLoading: isLoading(),
-    temporaryLayers: !_.isEmpty(getTemporaryLayers(state)),
+    hasLayerSelected: !!selectedLayer,
+    isLoading: selectedLayer && selectedLayer.isLayerLoading(),
     mapColors: getMapColors(state),
   };
 }
 
 function mapDispatchToProps(dispatch) {
   return {
-    closeFlyout: () => {
+    closeFlyout: layerId => {
       dispatch(updateFlyout(FLYOUT_STATE.NONE));
-      dispatch(clearTemporaryLayers());
+      dispatch(removeTransientLayer());
+      dispatch(setSelectedLayer(null));
+      dispatch(removeLayer(layerId));
     },
-    previewLayer: (layer) => {
+    previewLayer: layer => {
       dispatch(addLayer(layer.toLayerDescriptor()));
+      dispatch(setSelectedLayer(layer.getId()));
+      dispatch(setTransientLayer(layer.getId()));
+    },
+    removeLayer: () => {
```

Comment:
while we're here, I'd rename to `removeTransientLayer`, so we can better distinguish it from the `removeLayer` in the footer.